File: app/src/main/java/com/example/vn008xw/reddit/ui/best/BestFragment.java
package com.example.vn008xw.reddit.ui.best;

import android.content.Context;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.v7.widget.LinearLayoutManager;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.example.vn008xw.reddit.data.vo.RedditDataChild;
import com.example.vn008xw.reddit.databinding.FragmentBestBinding;
import com.example.vn008xw.reddit.ui.base.BaseFragment;

import java.util.List;

/**
 * Created by vn008xw on 11/17/17.
 */

public class BestFragment extends BaseFragment<BestContract.Presenter>
        implements BestContract.View {

    @Nullable
    private FragmentBestBinding mBinding;
    @Nullable
    private RedditListAdapter mAdapter;

    public static BestFragment newInstance() {
        return new BestFragment();
    }

    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater,
                             @Nullable ViewGroup container,
                             @Nullable Bundle savedInstanceState) {
        mBinding = FragmentBestBinding.inflate(inflater, container, false);
        return mBinding.getRoot();
    }


    @Override
    public void onActivityCreated(@Nullable Bundle savedInstanceState) {
        super.onActivityCreated(savedInstanceState);
        mAdapter = new RedditListAdapter(url -> {
            getPresenter().openImage(url);
        });
        mBinding.recyclerView.setAdapter(mAdapter);
        getPresenter().getPosts("");
    }

    @Override
    public void showLoading(boolean isLoading) {
        mBinding.setIsLoading(isLoading);
    }

    @Override
    public void showEntries(@NonNull List<RedditDataChild> entries) {
        if (mAdapter != null) {
            mAdapter.replace(entries);
        }
    }

    @Override
    public void showImage(@NonNull String thumbnail) {

    }
}

File: app/src/main/java/com/example/vn008xw/reddit/ui/best/BestFragment.kt
package com.example.vn008xw.reddit.ui.best

import android.content.Context
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.example.vn008xw.reddit.R
import com.example.vn008xw.reddit.data.vo.Entry
import com.example.vn008xw.reddit.data.vo.Image
import com.example.vn008xw.reddit.ui.base.BaseFragment


class BestFragment : BaseFragment<BestContract.Presenter>(), BestContract.View {

  override fun onCreate(savedInstanceState: Bundle?) {
    super.onCreate(savedInstanceState)
  }

  override fun onCreateView(inflater: LayoutInflater,
                            container: ViewGroup?,
                            savedInstanceState: Bundle?): View? {
    return inflater.inflate(R.layout.fragment_best, container, false)
  }

  override fun onAttach(context: Context?) {
    super.onAttach(context)
  }

  override fun onDetach() {
    super.onDetach()
  }

  override fun showLoading(isLoading: Boolean) {

  }

  override fun showEntries(entries: List<Entry>) {

  }

  override fun showImage(image: Image) {

  }

  companion object {
    @JvmStatic
    fun newInstance(): BestFragment {
      val fragment = BestFragment()
      val args = Bundle()
      fragment.arguments = args
      return fragment
    }
  }
}
